chia/types/spend_bundle_conditions.py

```diff
@@ -0,0 +1,29 @@
+from typing import List, Optional, Tuple
+from dataclasses import dataclass
+from chia.types.blockchain_format.sized_bytes import bytes32, bytes48
+from chia.util.ints import uint64, uint32
+from chia.util.streamable import Streamable, streamable
+
+
+# the Spend and SpendBundleConditions classes are mirrors of native types, returned by
+# run_generator2
+@dataclass(frozen=True)
+@streamable
+class Spend(Streamable):
+    coin_id: bytes32
+    puzzle_hash: bytes32
+    height_relative: Optional[uint32]
+    seconds_relative: uint64
+    create_coin: List[Tuple[bytes32, uint64, bytes]]
+    agg_sig_me: List[Tuple[bytes48, bytes]]
```

Comment:
I agree with Arvid that deferring is reasonable. I've thought a lot about the issues of parsing and rust and python data structures, and I believe in a lot of cases we want to do "dumb" deserialization since there may be instances where we don't want to validate anything (such as getting blobs off disk, which have presumably already been validated), and keeping `G1Element` objects as `bytes48` makes it clear that it's dumb deserialization that's been done and users have to ensure validation is done elsewhere.

In particular, when we start doing deserialization in rust, there's no easy way to get access to `G1Element` as either a python or C++ struct (and deserializing as a python object means grabbing the GIL which now ties the rust API to python in an unacceptable way), so keeping it around as `bytes48` is an obvious choice.